### PR TITLE
Implement round function

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

**Comment by okuta** [comment]
jenkins, test this please.

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -2118,6 +2126,62 @@ cdef _argmax = create_reduction_func(
...
+  }
+  return x;
+}'''
+
+
+cdef _round_float = '''
+if (in1 == 0) {
+    out0 = round(in0);
+} else {
+    long long x = pow10(abs(in1));
```
Comment (toslunar) [inline cupy/core/core.pyx]: The range of `long long` seems too small to support the following.
```
>>> np.around(1.5e200, decimals=-200)
1.0000000000000003e+200
>>> np.around(1.5e200, decimals=-199)
1.5000000000000005e+200
>>> np.around(1.5e200, decimals=-201)
0.0
```

Comment (okuta) [reply cupy/core/core.pyx]: Thanks, I fixed.

**Comment by asi1024** [comment]
Could you add a test to `cupy_tests/core_tests/test_fusion.py`?

File: cupy/core/core.pyx [hunk 2]
```diff
@@ -2118,6 +2126,63 @@ cdef _argmax = create_reduction_func(
...
+  return x;
+};
+'''
+
+
+cdef _round_float = '''
+if (in1 == 0) {
+    out0 = round(in0);
+} else {
+    double x = pow10<double>(abs(in1));
```
Comment (asi1024) [inline cupy/core/core.pyx]: Should not calculate this in the loop body.

Comment (okuta) [reply cupy/core/core.pyx]: I added

**Comment by okuta** [comment]
I rebased.
jenkins, test this please.

File: cupy/core/core.pyx [hunk 3]
```diff
@@ -2170,6 +2178,64 @@ cdef _argmax = create_reduction_func(
...
+};
+'''
+
+
+cdef _round_float = '''
+if (in1 == 0) {
+    out0 = round(in0);
+} else {
+    double x;
+    x = pow10<double>(abs(in1));  // This is placed outside loop by compiler.
```
Comment (asi1024) [inline cupy/core/core.pyx]: This line is NOT placed outside loop by compiler.

**Comment by okuta** [comment]
I rebased and updated comments.

**Comment by okuta** [comment]
I will add test.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**